source/UninstallTools/Factory/ScoopFactory.cs

```diff
@@ -92,59 +126,172 @@ public IList<ApplicationUninstallerEntry> GetUninstallerEntries(ListGenerationPr
+            if (Directory.Exists(installDir))
+            {
+                List<string> executables = new();
+                var currentDir = Path.Combine(installDir, "current");
+
+                try
                 {
-                    name = str;
+                    var install = JsonSerializer.Deserialize(
+                        File.ReadAllText(Path.Combine(currentDir, "install.json")),
+                        _jsonContext.AppInstall);
+
+                    var manifest = JsonSerializer.Deserialize(
+                        File.ReadAllText(Path.Combine(currentDir, "manifest.json")),
+                        _jsonContext.AppManifest);
+
+                    entry.AboutUrl = manifest.Homepage;
+
+                    var shortcuts = manifest.Architecture?[install.Architecture]?.Shortcuts ?? manifest.Shortcuts;
+                    if (shortcuts != null)
+                    {
+                        executables.AddRange(
+                            shortcuts.Select(x => Path.Combine(currentDir, x[0])));
+
+                        var icons = shortcuts
+                            .Where(x => x.Length >= 4 && x[3].EndsWith(".ico"))
+                            .Select(x => Path.Combine(currentDir, x[3]));
+                        if (icons.Any())
+                        {
+                            try
+                            {
+                                entry.IconBitmap = new Icon(icons.First());
+                            }
+                            catch { }
+                        }
+                    }
+
+                    var bin = manifest.Architecture?[install.Architecture]?.Bin ?? manifest.Bin;
+                    if (bin != null)
+                    {
+                        executables.AddRange(bin.Select(x => Path.Combine(installDir, "current", x)));
+                    }
+
+                    var env = manifest.Architecture?[install.Architecture]?.EnvAddPath ?? manifest.EnvAddPath;
+                    if (env != null)
+                    {
+                        currentDir = Path.Combine(currentDir, env[0]);
+                    }
                 }
+                catch (IOException)
+                { }
+                catch (UnauthorizedAccessException)
+                { }
+                catch (JsonException)
+                { }
 
-                var entry = new ApplicationUninstallerEntry
+                if (executables.Any())
                 {
-                    RawDisplayName = name,
-                    DisplayVersion = ApplicationEntryTools.CleanupDisplayVersion(version),
-                    RatingId = "Scoop " + name
-                };
-
-                var installDir = Path.Combine(isGlobal ? _scoopGlobalPath : _scoopUserPath, "apps\\" + name);
-                if (Directory.Exists(installDir))
+                    entry.SortedExecutables = executables.Distinct().ToArray();
```

Comment:
SortedExecutables should be sorted or some unexpected things could happen later.
```suggestion
                    entry.SortedExecutables = AppExecutablesSearcher.SortListExecutables(
                            executables.Distinct().Select(x => new FileInfo(x)),
                            entry.DisplayNameTrimmed)
                        .Select(x => x.FullName)
                        .ToArray();
```